tests/Silverback.Integration.Tests.E2E/Kafka/ErrorHandlingTests.cs

```diff
@@ -1177,5 +1265,81 @@ public async Task MovePolicy_ToOtherTopicAfterRetry_MessageRetriedAndMoved()
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://e2e";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .OnError(
+                                            policy => policy.MoveToKafkaTopic(
+                                                moveEndpoint => moveEndpoint.ProduceTo("other-topic")))
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                            })))
+                        .AddIntegrationSpy()
+                        .AddDelegateSubscriber(
+                            (IIntegrationEvent _) => throw new InvalidOperationException("Move!")))
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            await publisher.PublishAsync(message);
+
+            await Helper.WaitUntilAllMessagesAreConsumedAsync();
+
+            Helper.Spy.OutboundEnvelopes.Should().HaveCount(2);
+            Helper.Spy.InboundEnvelopes.Should().HaveCount(1);
+
+            Helper.Spy.OutboundEnvelopes[1].Message.Should()
+                .BeEquivalentTo(Helper.Spy.OutboundEnvelopes[0].Message);
+            Helper.Spy.OutboundEnvelopes[1].Endpoint.Name.Should().Be("other-topic");
+            Helper.Spy.OutboundEnvelopes[1].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.FailedAttempts,
+                        1));
+            Helper.Spy.OutboundEnvelopes[1].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.InvalidReason,
+                        "Exception has been thrown by the target of an invocation."));
+            Helper.Spy.OutboundEnvelopes[1].Headers
+                .Count(header => header.Name == DefaultMessageHeaders.SourceTimestamp)
+                .Should().Be(1);
+            Helper.Spy.OutboundEnvelopes[1].Headers
+                .Any(header => header.Name == DefaultMessageHeaders.TimestampKey)
+                .Should().BeFalse();
+            Helper.Spy.OutboundEnvelopes[1].Headers
+                .Count(header => header.Name == DefaultMessageHeaders.SourcePartition)
+                .Should().Be(1);
+            Helper.Spy.OutboundEnvelopes[1].Headers
```

Comment:
This isn't probably super reliable for a test, since the default value of an int is `0`. Maybe produce a couple of messages and see that the last one has the actual expected offset?

Reply:
I agree, I adjusted the e2e test.